core/classes/Core/Cache.php

```diff
@@ -65,14 +65,20 @@ public function isCached(string $key): bool {
             if (isset($cachedData[$key])) {
                 $entry = $cachedData[$key];
                 if ($entry && $this->_checkExpired($entry['time'], $entry['expire'])) {
-                    return false;
+                    $is_cached = false;
+                } else {
+                    $is_cached = isset($cachedData[$key]['data']);
                 }
-
-                return isset($cachedData[$key]['data']);
             }
         }
 
-        return false;
+        if (!isset($is_cached)) {
+            $is_cached = false;
+        }
+
+        CacheCollector::getInstance()->recordCheck($key, $is_cached);
```

Comment:
Can we be sure that, if we call this method, we won't see any errors in production websites (i.e. not using developer dependencies) because `CacheCollector` imports `DebugBar`